File: tests/expected/int_enum.go
package main

import (
	"fmt"
)

type Colors int

const (
	RED   Colors = iota
	GREEN        = iota
	BLUE         = iota
)

type Permissions int

const (
	R Permissions = 1
	W             = 2
	X             = 16
)

func Show() {
	color_map := map[Colors]string{RED: "red", GREEN: "green", BLUE: "blue"}
	var a Colors = GREEN
	if a == GREEN {
		fmt.Printf("%v\n", "green")
	} else {
		fmt.Printf("%v\n", "Not green")
	}
	var b Permissions = R
	if b == R {
		fmt.Printf("%v\n", "R")
	} else {
		fmt.Printf("%v\n", "Not R")
	}
	if !(len(color_map) != 0) {
		panic("assert")
	}
}

func main() {
	Show()
}

File: tests/expected/int_enum.jl
using SuperEnum

abstract type AbstractColors <: IntEnum end
abstract type AbstractPermissions <: IntFlag end
@se Colors::Int64 begin end
@se Permissions::Int64 begin end
function show()
    color_map = Dict(Colors.RED => "red", Colors.GREEN => "green", Colors.BLUE => "blue")
    a = Colors.GREEN
    if a == Colors.GREEN
        println("green")
    else
        println("Not green")
    end
    b = Permissions.R
    if b == Permissions.R
        println("R")
    else
        println("Not R")
    end
    @assert(length(color_map) != 0)
end

if abspath(PROGRAM_FILE) == @__FILE__
    show()
end
